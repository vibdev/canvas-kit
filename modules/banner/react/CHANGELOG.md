# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.25](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.24...@workday/canvas-kit-react-banner@0.1.25) (2019-06-20)


### Bug Fixes

* broad stroke accessibility fixes ([9803412](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/commits/9803412))





## [0.1.24](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.23...@workday/canvas-kit-react-banner@0.1.24) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.23](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.22...@workday/canvas-kit-react-banner@0.1.23) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.22](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.21...@workday/canvas-kit-react-banner@0.1.22) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.21](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.20...@workday/canvas-kit-react-banner@0.1.21) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.20](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.19...@workday/canvas-kit-react-banner@0.1.20) (2019-06-14)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.19](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.18...@workday/canvas-kit-react-banner@0.1.19) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.18](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.17...@workday/canvas-kit-react-banner@0.1.18) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.17](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.16...@workday/canvas-kit-react-banner@0.1.17) (2019-06-10)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.16](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.15...@workday/canvas-kit-react-banner@0.1.16) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.15](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.14...@workday/canvas-kit-react-banner@0.1.15) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.14](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.13...@workday/canvas-kit-react-banner@0.1.14) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.13](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.12...@workday/canvas-kit-react-banner@0.1.13) (2019-05-31)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.12](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.11...@workday/canvas-kit-react-banner@0.1.12) (2019-05-24)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.11](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.10...@workday/canvas-kit-react-banner@0.1.11) (2019-05-22)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.10](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.9...@workday/canvas-kit-react-banner@0.1.10) (2019-05-21)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.9](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.8...@workday/canvas-kit-react-banner@0.1.9) (2019-05-15)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.7...@workday/canvas-kit-react-banner@0.1.8) (2019-05-14)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.7](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.6...@workday/canvas-kit-react-banner@0.1.7) (2019-05-10)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.5...@workday/canvas-kit-react-banner@0.1.6) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.5](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.4...@workday/canvas-kit-react-banner@0.1.5) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.4](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.3...@workday/canvas-kit-react-banner@0.1.4) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.2...@workday/canvas-kit-react-banner@0.1.3) (2019-05-07)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.1...@workday/canvas-kit-react-banner@0.1.2) (2019-05-06)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## 0.1.1 (2019-04-30)

**Note:** Version bump only for package @workday/canvas-kit-react-banner
