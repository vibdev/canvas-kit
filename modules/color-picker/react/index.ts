import ColorInput from './lib/ColorInput';
import ColorPreview from './lib/ColorPreview';

export default ColorInput;
export {ColorInput, ColorPreview};
export * from './lib/ColorInput';
export * from './lib/ColorPreview';
