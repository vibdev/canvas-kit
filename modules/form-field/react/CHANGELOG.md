# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.4.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.4.1...@workday/canvas-kit-react-form-field@0.4.2) (2019-07-15)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.4.1](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.4.0...@workday/canvas-kit-react-form-field@0.4.1) (2019-06-20)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





# [0.4.0](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.14...@workday/canvas-kit-react-form-field@0.4.0) (2019-06-19)


### Features

* **Select:** fix from feedback and new folder structure ([cc24691](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/cc24691))
* **Select:** move stories to form field and add to packages ([564d33d](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/564d33d))
* **Select:** resolve conflicts ([48f7d17](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/48f7d17))





## [0.3.14](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.13...@workday/canvas-kit-react-form-field@0.3.14) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.13](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.12...@workday/canvas-kit-react-form-field@0.3.13) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.12](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.11...@workday/canvas-kit-react-form-field@0.3.12) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.11](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.10...@workday/canvas-kit-react-form-field@0.3.11) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.10](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.9...@workday/canvas-kit-react-form-field@0.3.10) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.9](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.8...@workday/canvas-kit-react-form-field@0.3.9) (2019-06-14)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.7...@workday/canvas-kit-react-form-field@0.3.8) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.7](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.6...@workday/canvas-kit-react-form-field@0.3.7) (2019-06-10)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.5...@workday/canvas-kit-react-form-field@0.3.6) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.5](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.4...@workday/canvas-kit-react-form-field@0.3.5) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.4](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.3...@workday/canvas-kit-react-form-field@0.3.4) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.2...@workday/canvas-kit-react-form-field@0.3.3) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.1...@workday/canvas-kit-react-form-field@0.3.2) (2019-05-31)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.3.1](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.3.0...@workday/canvas-kit-react-form-field@0.3.1) (2019-05-24)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





# [0.3.0](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.2.3...@workday/canvas-kit-react-form-field@0.3.0) (2019-05-23)


### Features

* **ColorPreview:** creating component ([368e189](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/368e189))
* **ColorPreview:** update from feedback ([9ced410](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/9ced410))





## [0.2.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.2.2...@workday/canvas-kit-react-form-field@0.2.3) (2019-05-22)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.2.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.2.1...@workday/canvas-kit-react-form-field@0.2.2) (2019-05-22)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.2.1](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.2.0...@workday/canvas-kit-react-form-field@0.2.1) (2019-05-21)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





# [0.2.0](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.1.7...@workday/canvas-kit-react-form-field@0.2.0) (2019-05-17)


### Features

* **ColorInput:** exclude story from tsconfig ([4d1b875](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/4d1b875))
* **ColorInput:** use TextInput ([b292d10](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/b292d10))





## [0.1.7](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.1.6...@workday/canvas-kit-react-form-field@0.1.7) (2019-05-15)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.1.6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.1.5...@workday/canvas-kit-react-form-field@0.1.6) (2019-05-14)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.1.5](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.1.4...@workday/canvas-kit-react-form-field@0.1.5) (2019-05-10)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.1.4](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.1.3...@workday/canvas-kit-react-form-field@0.1.4) (2019-05-09)


### Bug Fixes

* **form-field:** Fix type error in test ([69aed78](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/69aed78))





## [0.1.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.1.2...@workday/canvas-kit-react-form-field@0.1.3) (2019-05-09)


### Bug Fixes

* **common:** move storybook utils to outside folder ([1f661b6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/1f661b6))





## [0.1.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.1.1...@workday/canvas-kit-react-form-field@0.1.2) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





## [0.1.1](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/compare/@workday/canvas-kit-react-form-field@0.1.0...@workday/canvas-kit-react-form-field@0.1.1) (2019-05-07)

**Note:** Version bump only for package @workday/canvas-kit-react-form-field





# 0.1.0 (2019-05-06)


### Bug Fixes

* **form-field:** apply uniform height to input elements ([76113ca](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/76113ca))
* **form-field:** Clean up label interface ([86f13a8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/86f13a8))
* **form-field:** Show errors in the right places ([3544cea](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/3544cea))


### Features

* **common:** Add checkbox support to ControlledComponentWrapper ([54c78fb](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/54c78fb))
* **form-field:** Add aria-invalid if error ([b353aa9](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/b353aa9))
* **form-field:** add error prop to children ([e12d1f6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/e12d1f6))
* **form-field:** add grow prop to children ([cd94681](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/cd94681))
* **form-field:** Add support for aria-describedby ([67a545e](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/67a545e))
* **form-field:** Add support for htmlFor ([ebaf4c1](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/ebaf4c1))
* **form-field:** Add text area to form field stories ([087ed43](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/087ed43))
* **form-field:** Use fieldset and legend for RadioGroups ([48755eb](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-input-common/commits/48755eb))
