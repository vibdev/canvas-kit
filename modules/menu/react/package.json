{
  "name": "@workday/canvas-kit-react-menu",
  "version": "0.0.24",
  "description": "A container for navigation or action items",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts"
  ],
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build": "npm-run-all --parallel build:cjs build:es6"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "menu"
  ],
  "devDependencies": {
    "@material-ui/core": "^3.9.2",
    "@types/react-transition-group": "^2.0.14",
    "@workday/canvas-system-icons-web": "^1.0.1",
    "react-transition-group": "^2.5.0"
  },
  "dependencies": {
    "@types/uuid": "^3.4.4",
    "@workday/canvas-kit-react-card": "^0.3.23",
    "@workday/canvas-kit-react-common": "^0.7.5",
    "@workday/canvas-kit-react-core": "^1.10.6",
    "@workday/canvas-kit-react-icon": "^1.7.7",
    "@workday/design-assets-types": "^0.2.3",
    "emotion": "^9.2.12",
    "react-emotion": "^9.2.12",
    "uuid": "^3.3.2"
  },
  "peerDependencies": {
    "react": ">= 16.3 < 17"
  }
}
