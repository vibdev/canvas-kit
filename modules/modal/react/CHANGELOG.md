# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/modal/compare/@workday/canvas-kit-react-modal@0.0.1...@workday/canvas-kit-react-modal@0.0.2) (2019-06-20)

**Note:** Version bump only for package @workday/canvas-kit-react-modal





## 0.0.1 (2019-06-20)

**Note:** Version bump only for package @workday/canvas-kit-react-modal
