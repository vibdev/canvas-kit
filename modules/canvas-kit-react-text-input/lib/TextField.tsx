import * as React from 'react';
import styled from 'react-emotion';
import {GrowthBehavior} from '@workday/canvas-kit-react-common';
import {
  Label,
  LabelPosition,
  ErrorType,
  LabelPositionBehavior,
} from '@workday/canvas-kit-react-input-common';
import FormGroup from './FormGroup';
import TextInput, {TextInputProps} from './TextInput';
import Hint from './Hint';

export interface TextFieldProps extends TextInputProps {
  label?: React.ReactNode;
  hintText?: React.ReactNode;
}

const TextFieldInputContainer = styled('div')<GrowthBehavior & LabelPositionBehavior>(
  ({grow, labelPosition}) => {
    if (grow) {
      if (labelPosition === LabelPosition.Left) {
        return {
          flexGrow: 1,
        };
      }

      return {
        display: 'block',
      };
    }

    return {
      display: 'inline-block',
    };
  }
);

export default class TextField extends React.Component<TextFieldProps> {
  static LabelPosition = LabelPosition;
  static ErrorType = ErrorType;

  static defaultProps = {
    labelPosition: TextField.LabelPosition.Top,
  };

  render() {
    const {label, hintText, grow, ...inputProps} = this.props;
    const {labelPosition, error} = inputProps;

    return (
      <FormGroup labelPosition={labelPosition}>
        {typeof label === 'string' ? <Label labelPosition={labelPosition}>{label}</Label> : label}
        <TextFieldInputContainer grow={grow} labelPosition={labelPosition}>
          <TextInput grow={grow} {...inputProps} />
          {hintText && <Hint error={error}>{hintText}</Hint>}
        </TextFieldInputContainer>
      </FormGroup>
    );
  }
}
