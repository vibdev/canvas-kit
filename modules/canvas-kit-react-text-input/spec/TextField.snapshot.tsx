import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TextField from '../lib/TextField';

describe('TextField Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<TextField />);
    expect(component).toMatchSnapshot();
  });

  test('grow', () => {
    const component = renderer.create(<TextField grow={true} />);
    expect(component).toMatchSnapshot();
  });

  test('grow with left label', () => {
    const component = renderer.create(
      <TextField labelPosition={TextField.LabelPosition.Left} grow={true} />
    );
    expect(component).toMatchSnapshot();
  });
});
