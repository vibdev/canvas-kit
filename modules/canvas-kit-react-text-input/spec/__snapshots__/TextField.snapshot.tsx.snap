// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Snapshots grow 1`] = `
.emotion-6 {
  margin-bottom: 16px;
}

.emotion-4 {
  display: block;
}

.emotion-2 {
  display: block;
  position: relative;
}

.emotion-0 {
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  border: 1px solid #a1aab3;
  display: block;
  background-color: #ffffff;
  border-radius: 4px;
  box-sizing: border-box;
  height: 40px;
  min-width: 280px;
  -webkit-transition: 0.2s box-shadow,0.2s border-color;
  transition: 0.2s box-shadow,0.2s border-color;
  padding: 8px;
  width: 100%;
}

.emotion-0::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-0::-moz-placeholder {
  color: #5e6a75;
}

.emotion-0:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-0::placeholder {
  color: #5e6a75;
}

.emotion-0:hover {
  border-color: #7b858f;
}

.emotion-0:focus:not([disabled]) {
  border-color: #0875e1;
  box-shadow: inset 0 0 0 1px #0875e1;
  outline: none;
}

.emotion-0:disabled {
  background-color: #f6f7f8;
  border-color: #a1aab3;
  color: #a1aab3;
}

.emotion-0:disabled::-webkit-input-placeholder {
  color: #a1aab3;
}

.emotion-0:disabled::-moz-placeholder {
  color: #a1aab3;
}

.emotion-0:disabled:-ms-input-placeholder {
  color: #a1aab3;
}

.emotion-0:disabled::placeholder {
  color: #a1aab3;
}

<div
  className="emotion-6 emotion-7"
>
  <div
    className="emotion-4 emotion-5"
  >
    <div
      className="emotion-2 emotion-3"
    >
      <input
        className="emotion-0 emotion-1"
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`TextField Snapshots grow with left label 1`] = `
.emotion-2 {
  display: block;
  position: relative;
}

.emotion-0 {
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  border: 1px solid #a1aab3;
  display: block;
  background-color: #ffffff;
  border-radius: 4px;
  box-sizing: border-box;
  height: 40px;
  min-width: 280px;
  -webkit-transition: 0.2s box-shadow,0.2s border-color;
  transition: 0.2s box-shadow,0.2s border-color;
  padding: 8px;
  width: 100%;
}

.emotion-0::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-0::-moz-placeholder {
  color: #5e6a75;
}

.emotion-0:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-0::placeholder {
  color: #5e6a75;
}

.emotion-0:hover {
  border-color: #7b858f;
}

.emotion-0:focus:not([disabled]) {
  border-color: #0875e1;
  box-shadow: inset 0 0 0 1px #0875e1;
  outline: none;
}

.emotion-0:disabled {
  background-color: #f6f7f8;
  border-color: #a1aab3;
  color: #a1aab3;
}

.emotion-0:disabled::-webkit-input-placeholder {
  color: #a1aab3;
}

.emotion-0:disabled::-moz-placeholder {
  color: #a1aab3;
}

.emotion-0:disabled:-ms-input-placeholder {
  color: #a1aab3;
}

.emotion-0:disabled::placeholder {
  color: #a1aab3;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 24px;
}

.emotion-4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className="emotion-6 emotion-7"
>
  <div
    className="emotion-4 emotion-5"
  >
    <div
      className="emotion-2 emotion-3"
    >
      <input
        className="emotion-0 emotion-1"
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`TextField Snapshots renders as expected 1`] = `
.emotion-6 {
  margin-bottom: 16px;
}

.emotion-4 {
  display: inline-block;
}

.emotion-2 {
  display: inline-block;
  position: relative;
}

.emotion-0 {
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  border: 1px solid #a1aab3;
  display: block;
  background-color: #ffffff;
  border-radius: 4px;
  box-sizing: border-box;
  height: 40px;
  min-width: 280px;
  -webkit-transition: 0.2s box-shadow,0.2s border-color;
  transition: 0.2s box-shadow,0.2s border-color;
  padding: 8px;
}

.emotion-0::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-0::-moz-placeholder {
  color: #5e6a75;
}

.emotion-0:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-0::placeholder {
  color: #5e6a75;
}

.emotion-0:hover {
  border-color: #7b858f;
}

.emotion-0:focus:not([disabled]) {
  border-color: #0875e1;
  box-shadow: inset 0 0 0 1px #0875e1;
  outline: none;
}

.emotion-0:disabled {
  background-color: #f6f7f8;
  border-color: #a1aab3;
  color: #a1aab3;
}

.emotion-0:disabled::-webkit-input-placeholder {
  color: #a1aab3;
}

.emotion-0:disabled::-moz-placeholder {
  color: #a1aab3;
}

.emotion-0:disabled:-ms-input-placeholder {
  color: #a1aab3;
}

.emotion-0:disabled::placeholder {
  color: #a1aab3;
}

<div
  className="emotion-6 emotion-7"
>
  <div
    className="emotion-4 emotion-5"
  >
    <div
      className="emotion-2 emotion-3"
    >
      <input
        className="emotion-0 emotion-1"
        type="text"
      />
    </div>
  </div>
</div>
`;
